### PR TITLE
Change the code related to `pydrive` and use `google-api` instead

### LINKED ISSUE #2238: Use PyDrive2 instead of PyDrive
### Problem Description

The PyDrive library is deprecated and no longer maintained. We should shift to using [PyDrive2](https://github.com/iterative/PyDrive2) or the [google api client](https://github.com/googleapis/google-api-python-client) directly.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #2238 
This `pr` changes the usage of `pydrive` to official `google` api calls without the need of third party libraries like `PyDrive2` or the currently deprecated `PyDrive`.

CU-86b29t2nz

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b29t2nz SDV - Use PyDrive2 instead of PyDrive #2238](https://app.clickup.com/t/86b29t2nz)

**Review by amontanez24 — CHANGES_REQUESTED** [review]
I think we can simplify the authentication section. We never authenticate by loading from a json file on the user's machine so I don't think we need that option. We can either just refresh the credentials like we do or use InstalledAppFlow.from_client_config and just load from the string

File: tests/_external/gdrive_utils.py [hunk 1]
```diff
@@ -25,53 +26,38 @@ def _generate_filename():
...
+    tmp_credentials = os.getenv('PYDRIVE_CREDENTIALS')
+    creds = None
+    if tmp_credentials:
+        credentials = json.loads(tmp_credentials)
+        creds = Credentials.from_authorized_user_info(credentials, SCOPES)
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file('client_secrets.json', SCOPES)
```
[the remark this answers is not in the document]
Comment (pvk-developer) [reply tests/_external/gdrive_utils.py:38]: Oh sorry this was from my local. I will remove it

**Comment by pvk-developer** [comment]
> I think we can simplify the authentication section. We never authenticate by loading from a json file on the user's machine so I don't think we need that option. We can either just refresh the credentials like we do or use InstalledAppFlow.from_client_config and just load from the string

Yes I agree, I addressed this here:
https://github.com/sdv-dev/SDV/pull/2254/commits/8de9442f5ee4641776638403de566dddf9ac5c42

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Review by lajohn4747 — APPROVED** [review]

**Review by amontanez24 — APPROVED** [review]
LGTM! 🚢

**Pull request merged**